### PR TITLE
Cata make dead fish swim solution

### PR DESCRIPTION
**Pull request by mpsujek**

### CODE REVIEW

File: js/make_dead_fish_swim.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//https://www.codewars.com/kata/51e0007c1f9378fa810002a9/train/javascript
+
+function parse( data ){
+  const dataArray = data.split('');
+  const arrayReturn = [];
+  let element = 0;
+  dataArray.map(function(el){
```
Comment (Gelio) [inline js/make_dead_fish_swim.js]: Why use `map`?

File: js/make_dead_fish_swim.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    (el === 'i') && (element += 1);
```
Comment (Gelio) [inline js/make_dead_fish_swim.js]: Clever tricks 😄

File: js/make_dead_fish_swim.js (hunk 1 above)
Comment (mpsujek) [reply js/make_dead_fish_swim.js]: Just training

Comment (Gelio) [reply js/make_dead_fish_swim.js]: I mean why `map` and not sth like `forEach` :smile:

Comment (mpsujek) [reply js/make_dead_fish_swim.js]: I've added forEach soluton :)

Comment (Gelio) [reply js/make_dead_fish_swim.js]: My intention was to replace `map` with `forEach`, as `map` is not necessary here :smile:

Comment (mpsujek) [reply js/make_dead_fish_swim.js]: I know, but I wanted to do this, especially with a map. So due to your suggestion, I've done that also with forEach.

Comment (Gelio) [reply js/make_dead_fish_swim.js]: I think I don't understand.

* `map` should be used when you transform every element in some way and need an array of results.
* `forEach` is used when you want to invoke some logic for each element, but don't care about the array of results.

I would still suggest removing the solution with `map`, as seems unnecessary (not using the right tool for the job) :smile:

**Comment by Gelio** [comment]
Nice work :+1: Looks good

**Pull request merged**